packages/server-wallet/src/__chain-test__/direct-funding.test.ts

```diff
@@ -0,0 +1,128 @@
+import {CreateChannelParams, Participant, Allocation} from '@statechannels/client-api-schema';
+import {makeDestination} from '@statechannels/wallet-core';
+import {BigNumber, ethers} from 'ethers';
+
+import {defaultConfig} from '../config';
+import {NotificationReceiver, SingleChannelOutput, Wallet} from '../wallet';
+import {getChannelResultFor, getPayloadFor} from '../__test__/test-helpers';
+
+const b = new Wallet({...defaultConfig, postgresDBName: 'TEST_B'});
+const a = new Wallet({...defaultConfig, postgresDBName: 'TEST_A'});
+
+beforeAll(async () => {
+  await a.dbAdmin().createDB();
+  await b.dbAdmin().createDB();
+  await Promise.all([a.dbAdmin().migrateDB(), b.dbAdmin().migrateDB()]);
+});
+afterAll(async () => {
+  await Promise.all([a.destroy(), b.destroy()]);
+  await a.dbAdmin().dropDB();
+  await b.dbAdmin().dropDB();
+  a.destroy();
+  b.destroy();
+});
+
+it('Create a directly funded channel between two wallets ', async () => {
+  const participantA: Participant = {
+    signingAddress: await a.getSigningAddress(),
+    participantId: 'a',
+    destination: makeDestination(
+      '0xaaaa000000000000000000000000000000000000000000000000000000000001'
+    ),
+  };
+  const participantB: Participant = {
+    signingAddress: await b.getSigningAddress(),
+    participantId: 'b',
+    destination: makeDestination(
+      '0xbbbb000000000000000000000000000000000000000000000000000000000002'
+    ),
+  };
+
+  const allocation: Allocation = {
+    allocationItems: [
+      {
+        destination: participantA.destination,
+        amount: BigNumber.from(1).toHexString(),
+      },
+      {
+        destination: participantB.destination,
+        amount: BigNumber.from(0).toHexString(),
+      },
+    ],
+    token: '0x00', // must be even length
+  };
+
+  const channelParams: CreateChannelParams = {
+    participants: [participantA, participantB],
+    allocations: [allocation],
+    appDefinition: ethers.constants.AddressZero,
+    appData: '0x00', // must be even length
+    fundingStrategy: 'Direct',
+  };
+
+  let counter = 0;
+  const postFundAPromise = new Promise<SingleChannelOutput>(resolve => {
+    const aNotificationReceiver: NotificationReceiver = {
+      onWalletNotification(message: SingleChannelOutput) {
+        if (counter > 0) {
+          resolve(message);
+        }
+        counter++;
+      },
+    };
+    a.notificationReceiver = aNotificationReceiver;
+  });
```

Comment:
If you decided to go with the event emitter approach below, this would become a lot simpler. You could just do something like:
```ts
await new Promise(resolve => a.onNotification((channelResult) => resolve(channelResult))) // first notification
await new Promise(resolve => a.onNotification((channelResult) => resolve(channelResult))) // second notification
```

Reply:
I wonder if it is possible to miss the second notification with this approach. I am not sure.

Is it a concern that the second notification would fire before the second event listener is registered?

Reply:
Maybe that's not a concern as ["The EventEmitter calls all listeners synchronously"](https://nodejs.org/api/events.html#events_asynchronous_vs_synchronous)